client/state/audio/middleware.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Internal dependencies
+ */
+import {
+	HAPPYCHAT_RECEIVE_EVENT,
+} from 'state/action-types';
+
+const isAudioSupported = () => typeof window === 'object' && typeof window.Audio === 'function';
```

Comment:
This is a bit messy to stub.  Another alternative is to export an object. You can then update the reference there without checking these props or manipulating the window object.

```jsx
const export isSupported = { audio: typeof window === 'object' && typeof window.Audio === 'function' };
```

Then in the test you can do:
```
import middleware, { isSupported } from '../middleware';
// In your test setup
isSupported.audio = false;
```